contributors/design-proposals/grow-volume-size.md

```diff
@@ -0,0 +1,253 @@
+
+## Volume Plugin Matrix
+
+
+| Volume Plugin   | Supports Resize   | Requires File system Resize | Supported in 1.8 Release |
+| ----------------| :---------------: | :--------------------------:| :----------------------: |
+| EBS             | Yes               | Yes                         | Yes                      |
+| GCE PD          | Yes               | Yes                         | Yes                      |
+| GlusterFS       | Yes               | No                          | Yes                      |
+| Cinder          | Yes               | Yes                         | Yes                      |
+| Vsphere         | Yes               | Yes                         | No                       |
+| Ceph RBD        | Yes               | Yes                         | No                       |
+| Host Path       | No                | No                          | No                       |
+| Azure Disk      | Yes               | Yes                         | No                       |
+| Azure File      | No                | No                          | No                       |
+| Cephfs          | No                | No                          | No                       |
+| NFS             | No                | No                          | No                       |
+
+
+## Implementation Design
+
+For volume type that requires both file system expansion and a volume plugin based modification, growing persistent volumes will be two
+step process.
+
+
+For volume types that only require volume plugin based api call, this will be one step process.
+
+### Prerequisite
+
+* `pvc.spec.resources.requests.storage` field of pvc object will become mutable after this change.
+* #sig-api-machinery has agreed to allow pvc's status update from kubelet as long as pvc and node relationship
+  can be validated by node authorizer.
+* This feature will be protected by an alpha feature gate.
+
+### Admission Control and Validations
+
+* Resource quota code has to be updated to take into account PVC expand feature.
+* In case volume plugin doesn’t support resize feature. The resize API request will be rejected and PVC object will not be saved. This check will be performed via an admission controller plugin.
+* In case requested size is smaller than current size of PVC. A validation will be used to reject the API request. (This could be moved to admission controller plugin too.)
+
+
+### Controller Manager resize
+
+A new controller called `volume_expand_controller` will listen for pvc size expansion requests and take action as needed. The steps performed in this
+new controller will be:
+
+* Watch for pvc update requests and add pvc to controller's desired state of world if a increase in volume size was requested. Once PVC is added to
+  controller's desired state of world - `pvc.Status.Conditions` will be updated with `ResizeStarted: True`.
```

Comment:
Is the purpose of this field so that multiple resizes don't happen at the same time on the same volume?  What happens if the resize process crashes in the middle?